addons/project/models/project.py

```diff
@@ -451,6 +451,9 @@ def map_tasks(self, new_project_id):
             if task.parent_id:
                 # set the parent to the duplicated task
                 defaults['parent_id'] = old_to_new_tasks.get(task.parent_id.id, False)
+            elif task.display_project_id.id == self.id:
```

Comment:
Equivalent to:
```suggestion
            elif task.display_project_id == self:
```

Reply:
`.id` is used to have and ensure_one as this would not work with multiple projects but adding one might (and is) be cleaner